core/src/main/java/org/mapfish/print/http/HttpRequestCache.java

```diff
@@ -0,0 +1,178 @@
+package org.mapfish.print.http;
+
+
+import jsr166y.ForkJoinPool;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.AbstractClientHttpResponse;
+import org.springframework.http.client.ClientHttpRequest;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/**
+ * 
+ * Cached ClientHttpRequestFactory.
+ * 
+ * @author Niels Charlier
+ *
+ */
+public final class HttpRequestCache {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpRequestCache.class);
+    
+    private static class CachedClientHttpResponse extends AbstractClientHttpResponse {
+        
+        private final File cachedFile;
+        private final HttpHeaders headers;
+        private final int status;
+        private final String statusText;
+        
+        public CachedClientHttpResponse(final ClientHttpResponse originalResponse) throws IOException {
+            this.headers = originalResponse.getHeaders();
+            this.status = originalResponse.getRawStatusCode();
+            this.statusText = originalResponse.getStatusText();
+            this.cachedFile = File.createTempFile("cacheduri", null);
+            InputStream is = originalResponse.getBody();
+            try {
+                OutputStream os = new FileOutputStream(this.cachedFile);
+                try {
+                    IOUtils.copy(is, os);
+                } finally {
+                    os.close();
+                }
+            } finally {
+                is.close();
+            }
+        }
+
+        @Override
+        public InputStream getBody() throws IOException {
+            return new FileInputStream(this.cachedFile);
```

Comment:
It doesn't, it only closes the response. And after a quick look at other implementations, it appears that they do close the stream on closing the response. Double closing never hurts, so I'm going to change that here too.